tool/metrics_monitor.py

```diff
@@ -0,0 +1,75 @@
+import time
+import datetime as dt
+import threading
+import sys
+import psutil
+import firebase_admin
+from firebase_admin import credentials
+from firebase_admin import firestore
+from core_data_modules.logging import Logger
+
+COLLECTION = 'pipeline_metrics'
+DEFAULT_INTERVAL = 3
+
+log = Logger(__name__)
+
+
+def initialize_firebase(CRYPTO_TOKEN_PATH):
+    global firebase_client
+    log.info("Setting up Firebase client")
+    firebase_cred = credentials.Certificate(CRYPTO_TOKEN_PATH)
+    firebase_admin.initialize_app(firebase_cred)
+    firebase_client = firestore.client()
+    log.info("Firebase client ready")
+
+
+def get_metrics(interval):
```

Comment:
rename to `get_and_publish_system_metrics` ?

Reply:
I agree. I've changed it